src/PregWrapper.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer;
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
+ *
+ * @internal
+ */
+final class PregWrapper
+{
+    /**
+     * @param string   $pattern
+     * @param string   $subject
+     * @param string[] &$matches
+     * @param int      $flags
+     * @param int      $offset
+     *
+     * @throws \RuntimeException
```

Comment:
IMO we shall have dedicated exception thrown (one exception for all methods is enough)